idaes/core/solvers/petsc.py

```diff
@@ -517,10 +518,28 @@ def petsc_dae_by_time_element(
         between = pyo.Set(initialize=sorted(between))
         between.construct()
 
+    # Make sure that time is a ContinuousSet that has been discretized
+    try:
+        time.get_discretization_info()["nfe"]
+    except AttributeError:
+        raise RuntimeError("Argument time is not a Pyomo ContinuousSet")
+    except KeyError:
+        raise RuntimeError("The ContinuousSet time has not been discretized")
+
     solve_log = idaeslog.getSolveLogger("petsc-dae")
 
-    regular_vars, time_vars = flatten_dae_components(m, time, pyo.Var)
-    regular_cons, time_cons = flatten_dae_components(m, time, pyo.Constraint)
+    # Need a representative time for flatten_dae_components to work, because otherwise things like
+    # sums of mole fraction constraints being deactivated at t0 (because all the mole fractions are fixed),
+    # etc., cause flatten_dae_components to miss a constraint that is active at t>t0
+    if representative_time is None:
+        representative_time = between.at(2)  # Two because Pyomo sets start at one
```

Comment:
If `representative_time` is provided, can you make sure it is an element of `between`?